### PR TITLE
Use nunjucks to generate siteNavs and the site map

### PR DESCRIPTION
**Pull request by damithc**
## Before:
image
image

## After:
image
image

```
<include>ing the (ug|dg)SiteNave.md in the landing page site-map avoids
duplication but prevents optimal use of siteNav as well as site-map.
For example, we cannot use `:expanded:` in the siteNav to make an item
collapsible because `:expanded:` appears as raw text in the landing page.

Let's use nunjucks to define siteNav items as an array and a macro to
render it optimally for both reuse contexts.

Pros:
* siteNav items are now collapsible
* landing page shows links as a hierarchy rather than a flat list

Cons:
* The code is more complicated than before: But, the complexity is
  within the expertise of RepoSense developers as nunjucks code is
  similar to JavaScript.
```

### CODE REVIEW

**Comment by damithc** [comment]
Ready for review

File: docs/scripts/macros.njk [hunk 1]
```diff
@@ -9,6 +22,7 @@
 <p/>
 {% endmacro %}
 
+
```
Comment (Tejas2805) [inline docs/scripts/macros.njk:25]: Not sure about this but are two blank lines required?

**Comment by damithc** [comment]
> These changes can't be seen in the `Production Site`. Is it intended? Or is it due to different versions of MarkBind being used for the two sites?

There is no preview for the production site. You must have been looking at the actual production site which has not been updated yet with these changes.

File: docs/scripts/macros.njk (hunk 1 above)
Comment (damithc) [reply docs/scripts/macros.njk:25]: I usually use two blank lines between macros, similar to python convention of putting two blank lines between functions.

**Comment by damithc** [comment]
@jamessspanggg have a look and merge if no issues. Documentation PRs can be merged faster. If any issues are discovered later, we can always fix as a separate PR.

**Comment by damithc** [comment]
https://reposense.org/ updated

**Pull request merged**